Test fails when using unix line ends (\n) instead of windows line endings (\r\n)

Test fails when using unix line ends (\n) instead of windows line endings (\r\n)

I just forked and cloned the repo and run npm test. 

```
$ npm t

> comment-parser@0.5.2 pretest C:\repos\comment-parser
> eslint .


> comment-parser@0.5.2 test C:\repos\comment-parser
> mocha tests



  parse() with custom tag parsers
    √ should use `opts.parsers`
    √ should merge parsers result
    √ should catch parser exceptions and populate `errors` field

  File parsing
    √ should parse the file by path
    √ should path the error if file dows not exist
    √ should return `Transform` stream

  Comment string parsing
    √ should parse doc block with description
    √ should parse doc block with no mid stars
    √ should skip surrounding empty lines while preserving line numbers
    √ should accept a description on the first line
    √ should parse not starred middle lines with `opts.trim = true`
    1) should parse not starred middle lines with `opts.trim = false`
    √ should accept comment close on a non-empty
    √ should skip empty blocks
    √ should parse multiple doc blocks
    √ should parse one line block
    √ should skip `/* */` comments
    √ should skip `/*** */` comments
    2) should preserve empty lines and indentation with `opts.trim = false`
    √ should parse one line block with tag
    √ should parse `@tag`
    √ should parse `@tag {my.type}`
    √ should parse tag with name only `@tag name`
    √ should parse tag with type and name `@tag {my.type} name`
    √ should parse tag with type, name and description `@tag {my.type} name description`
    √ should parse tag with type, name and description `@tag {my.type} name description separated by tab`
    √ should parse tag with multiline description
    √ should parse tag with type and optional name `@tag {my.type} [name]`
    √ should parse tag with type and optional name with default value `@tag {my.type} [name=value]`
    √ should tolerate default value with whitespces `@tag {my.type} [name=John Doe]`
    √ should tolerate quoted default value `@tag [name="yay!"]`
    √ should keep value as is if quotes are mismatched `@tag [name="yay']`
    √ should parse rest names `@tag ...name desc`
    √ should parse optional rest names `@tag [...name] desc`
    √ should parse multiple tags
    √ should parse nested tags
    √ should parse complex types `@tag {{a: type}} name`
    √ should gracefully fail on syntax errors `@tag {{a: type} name`
    √ should parse $ in description`
    √ should parse doc block with bound forced to the left
    √ should parse doc block with bound forced to the right
    √ should parse doc block with soft bound
    3) should parse doc block with soft bound respecting `opts.trim = false`
    √ should parse multiline without star as same line respecting `opts.join = true`
    √ should parse multiline without star as same line respecting `opts.join = "\t"`
    4) should parse multiline without star as same line with intent respecting `opts.join = 1` and `opts.trim = false`
    √ should parse same line closing section (issue #22)
    5) should tolerate inconsistent formatting (issue #29)


  43 passing (91ms)
  5 failing

  1) Comment string parsing
       should parse not starred middle lines with `opts.trim = false`:

      AssertionError: expected { Object (tags, line, ...) } to deeply equal { Object (description, source, ...) }
      + expected - actual

       {
      -  "description": "\r\nDescription text\r"
      +  "description": "\nDescription text"
         "line": 1
      -  "source": "\r\nDescription text\r\n@tag tagname Tag description\r\n"
      +  "source": "\nDescription text\n@tag tagname Tag description\n"
         "tags": [
           {
      -      "description": "Tag description\r\n"
      +      "description": "Tag description\n"
             "line": 3
             "name": "tagname"
             "optional": false
      -      "source": "@tag tagname Tag description\r\n"
      +      "source": "@tag tagname Tag description\n"
             "tag": "tag"
             "type": ""
           }
         ]

      at Context.<anonymous> (tests\parse.spec.js:105:8)

  2) Comment string parsing
       should preserve empty lines and indentation with `opts.trim = false`:

      AssertionError: expected { Object (tags, line, ...) } to deeply equal { Object (description, source, ...) }
      + expected - actual

       {
      -  "description": "\r\n\n\n  Description first line\r\n    second line\r\n\n      third line\r\n"
      +  "description": "\n\n\n  Description first line\n    second line\n\n      third line\n"
         "line": 1
      -  "source": "\r\n\n\n  Description first line\r\n    second line\r\n\n      third line\r\n"
      +  "source": "\n\n\n  Description first line\n    second line\n\n      third line\n"
         "tags": []
       }

      at Context.<anonymous> (tests\parse.spec.js:227:8)

  3) Comment string parsing
       should parse doc block with soft bound respecting `opts.trim = false`:

      AssertionError: expected { Object (tags, line, ...) } to deeply equal { Object (description, source, ...) }
      + expected - actual

       {
      -  "description": "\r\nDescription text\r\n  another line\r"
      +  "description": "\nDescription text\n  another line"
         "line": 1
      -  "source": "\r\nDescription text\r\n  another line\r\n@tag tagname Tag description\r\n"
      +  "source": "\nDescription text\n  another line\n@tag tagname Tag description\n"
         "tags": [
           {
      -      "description": "Tag description\r\n"
      +      "description": "Tag description\n"
             "line": 4
             "name": "tagname"
             "optional": false
      -      "source": "@tag tagname Tag description\r\n"
      +      "source": "@tag tagname Tag description\n"
             "tag": "tag"
             "type": ""
           }
         ]

      at Context.<anonymous> (tests\parse.spec.js:797:11)

  4) Comment string parsing
       should parse multiline without star as same line with intent respecting `opts.join = 1` and `opts.trim = false`:

      AssertionError: expected { Object (tags, line, ...) } to deeply equal { Object (line, description, ...) }
      + expected - actual

       {
      -  "description": "\r"
      +  "description": ""
         "line": 1
      -  "source": "\r\n@tag name\r\ndescription\r\n  intent same line\r\n"
      +  "source": "\n@tag name\ndescription\n  intent same line\n"
         "tags": [
           {
      -      "description": "description\r  intent same line\r\n"
      +      "description": "description  intent same line\n"
             "line": 2
             "name": "name"
             "optional": false
      -      "source": "@tag name\r\ndescription\r  intent same line\r\n"
      +      "source": "@tag name\ndescription  intent same line\n"
             "tag": "tag"
             "type": ""
           }
         ]

      at Context.<anonymous> (tests\parse.spec.js:876:11)

  5) Comment string parsing
       should tolerate inconsistent formatting (issue #29):

      AssertionError: expected { Object (tags, line, ...) } to deeply equal { Object (description, line, ...) }
      + expected - actual

       {
      -  "description": "\r"
      +  "description": ""
         "line": 1
      -  "source": "\r\n@param {Object} options 配置\r\n@return {Any} obj 组件返回的对象\r\n@example\r\nvar widget = XX.showWidget('searchlist', {\r\n   onComplete: function
() {\r\n         todoSomething();\r\n    }\r\n});\r\n"
      +  "source": "\n@param {Object} options 配置\n@return {Any} obj 组件返回的对象\n@example\nvar widget = XX.showWidget('searchlist', {\n   onComplete: function() {\n
     todoSomething();\n    }\n});\n"
         "tags": [
           {
      -      "description": "配置\r"
      +      "description": "配置"
             "line": 2
             "name": "options"
             "optional": false
      -      "source": "@param {Object} options 配置\r"
      +      "source": "@param {Object} options 配置"
             "tag": "param"
             "type": "Object"
           }
           {
      -      "description": "组件返回的对象\r"
      +      "description": "组件返回的对象"
             "line": 3
             "name": "obj"
             "optional": false
      -      "source": "@return {Any} obj 组件返回的对象\r"
      +      "source": "@return {Any} obj 组件返回的对象"
             "tag": "return"
             "type": "Any"
           }
           {
      -      "description": "var widget = XX.showWidget('searchlist', {\r\n   onComplete: function() {\r\n         todoSomething();\r\n    }\r\n});\r\n"
      +      "description": "widget = XX.showWidget('searchlist', {\n   onComplete: function() {\n         todoSomething();\n    }\n});\n"
             "line": 4
      -      "name": "\r"
      +      "name": "\nvar"
             "optional": false
      -      "source": "@example\r\nvar widget = XX.showWidget('searchlist', {\r\n   onComplete: function() {\r\n         todoSomething();\r\n    }\r\n});\r\n"
      +      "source": "@example\nvar widget = XX.showWidget('searchlist', {\n   onComplete: function() {\n         todoSomething();\n    }\n});\n"
             "tag": "example"
             "type": ""
           }
         ]

      at Context.<anonymous> (tests\parse.spec.js:922:15)



npm ERR! Test failed.  See above for more details.


```

--- a/tests/parse.js
+++ b/tests/parse.js
@@ -11,8 +11,8 @@ var parse = require('../index')
 
 module.exports = function (func, opts) {
   var str = func.toString()
-  return parse(str.slice(
-    str.indexOf('{') + 1,
-    str.lastIndexOf('}')
-  ), opts)
+  str = str
+    .slice(str.indexOf('{') + 1, str.lastIndexOf('}'))
+    .replace(/\r\n/g, '\n')
+  return parse(str, opts)
 }

